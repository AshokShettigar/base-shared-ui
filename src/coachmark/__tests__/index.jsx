import { mount } from 'enzyme';
import { Coachmark } from 'platform-common-ui';

describe('coachmark', () => {
    const onTourEnd = jest.fn();
    const onStepNext = jest.fn();
    const onTourStart = jest.fn();
    const handleDocumentClick = jest.fn();

    const tour = {
        id: 'desktop-summary-tour',
        steps: [],
        scrollTopMargin: 100,
        bubblePadding: 0,
        arrowWidth: 10,
        customRenderer: 'rmm_bubble',
        wrapperClass: 'rmm-hopscotch',
        i18n: {
            prevBtn: 'Back',
            nextBtn: 'Next',
            doneBtn: 'Close',
            closeTooltip: 'Close'
        }
    };

    const steps = [
        {
            target: 'coachmark-step1',
            title: 'Welcome to CoachMarks!',
            content: 'Hey there! This is an example CoachMarks tour.',
            placement: 'bottom',
            arrowOffset: 60
        },
        {
            target: 'coachmark-step2',
            title: 'This is the title for step 2',
            content: 'Content for step 2 goes here',
            placement: 'left',
            showPrevButton: true,
            yOffset: 10,
            width: 235
        },
        {
            target: 'coachmark-step3',
            placement: 'bottom',
            title: 'This is the title for step 3',
            content: 'Content for step 3 goes here',
            arrowColor: '#0D3560',
            showPrevButton: true
        },
        {
            target: 'coachmark-step4',
            placement: 'right',
            title: 'Tour ends here',
            content: 'If you want to say anything, here is your last chance',
            showPrevButton: true
        }
    ];

    it('should render the coachmark tour', function() {
        const tree = mount(
            <Coachmark
                tour={tour}
                steps={steps}
                type="tour"
                showCoachmark
                onTourStart={onTourStart}
                onTourEnd={onTourEnd}
                onStepNext={onStepNext}
                handleDocumentClick={handleDocumentClick}
            />
        );
        expect(tree).toMatchSnapshot();
    });
});

