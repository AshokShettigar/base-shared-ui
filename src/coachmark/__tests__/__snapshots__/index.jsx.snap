// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coachmark should render the coachmark tour 1`] = `
<Coachmark
  handleDocumentClick={[Function]}
  onDocumentClick={[Function]}
  onStepNext={[Function]}
  onTourEnd={[Function]}
  onTourStart={[Function]}
  showCoachmark={true}
  steps={
    Array [
      Object {
        "arrowOffset": 60,
        "content": "Hey there! This is an example CoachMarks tour.",
        "placement": "bottom",
        "target": "coachmark-step1",
        "title": "Welcome to CoachMarks!",
      },
      Object {
        "content": "Content for step 2 goes here",
        "placement": "left",
        "showPrevButton": true,
        "target": "coachmark-step2",
        "title": "This is the title for step 2",
        "width": 235,
        "yOffset": 10,
      },
      Object {
        "arrowColor": "#0D3560",
        "content": "Content for step 3 goes here",
        "placement": "bottom",
        "showPrevButton": true,
        "target": "coachmark-step3",
        "title": "This is the title for step 3",
      },
      Object {
        "content": "If you want to say anything, here is your last chance",
        "placement": "right",
        "showPrevButton": true,
        "target": "coachmark-step4",
        "title": "Tour ends here",
      },
    ]
  }
  tour={
    Object {
      "arrowWidth": 10,
      "bubblePadding": 0,
      "customRenderer": "rmm_bubble",
      "i18n": Object {
        "closeTooltip": "Close",
        "doneBtn": "Close",
        "nextBtn": "Next",
        "prevBtn": "Back",
      },
      "id": "desktop-summary-tour",
      "onClose": [Function],
      "onEnd": [Function],
      "onNext": [Function],
      "onStart": [Function],
      "scrollTopMargin": 100,
      "steps": Array [
        Object {
          "arrowOffset": 60,
          "content": "Hey there! This is an example CoachMarks tour.",
          "placement": "bottom",
          "target": "coachmark-step1",
          "title": "Welcome to CoachMarks!",
        },
        Object {
          "content": "Content for step 2 goes here",
          "placement": "left",
          "showPrevButton": true,
          "target": "coachmark-step2",
          "title": "This is the title for step 2",
          "width": 235,
          "yOffset": 10,
        },
        Object {
          "arrowColor": "#0D3560",
          "content": "Content for step 3 goes here",
          "placement": "bottom",
          "showPrevButton": true,
          "target": "coachmark-step3",
          "title": "This is the title for step 3",
        },
        Object {
          "content": "If you want to say anything, here is your last chance",
          "placement": "right",
          "showPrevButton": true,
          "target": "coachmark-step4",
          "title": "Tour ends here",
        },
      ],
      "wrapperClass": "rmm-hopscotch",
    }
  }
  type="tour"
/>
`;
